File: src/com/sterndu/json/NullValue.java
package com.sterndu.json;

import java.util.function.Function;

public class NullValue implements JsonValue {

	public Object getValue() {
		return null;
	}

	@Override
	public String toJson() {
		return "null";
	}

	@Override
	public String toJson(Function<Object, String> function) {
		return "null";
	}

	@Override
	public JsonValue toJsonValue() {
		return this;
	}

	@Override
	public JsonValue toJsonValue(Function<Object, String> function) {
		return this;
	}

	@Override
	public String toString() {
		return toJson();
	}

}

File: src/com/sterndu/json/NullValue.kt
@file:JvmName("NullValue")
package com.sterndu.json

class NullValue : JsonValue {
	val value: Any?
		get() = null

	override fun toJson(): String {
		return "null"
	}

	override fun toJson(function: Function1<Any?, String>): String {
		return "null"
	}

	override fun toJsonValue(): JsonValue {
		return this
	}

	override fun toString(): String {
		return toJson()
	}

	override fun toJsonValue(function: Function1<Any?, String>): JsonValue {
		return this
	}
}
